lib/rust/ensogl/component/grid-view/src/lib.rs

```diff
@@ -0,0 +1,508 @@
+//! ListView EnsoGL Component.
+//!
+//! ListView a displayed list of entries with possibility of selecting one and "choosing" by
+//! clicking or pressing enter - similar to the HTML `<select>`.
+
+#![recursion_limit = "1024"]
+// === Features ===
+#![feature(option_result_contains)]
+#![feature(trait_alias)]
+#![feature(hash_drain_filter)]
+// === Standard Linter Configuration ===
+#![deny(non_ascii_idents)]
+#![warn(unsafe_code)]
+// === Non-Standard Linter Configuration ===
+#![warn(missing_copy_implementations)]
+#![warn(missing_debug_implementations)]
+#![warn(missing_docs)]
+#![warn(trivial_casts)]
+#![warn(trivial_numeric_casts)]
+#![warn(unused_import_braces)]
+#![warn(unused_qualifications)]
+
+
+// ==============
```

Comment:
spacing - is this layout with our auto-formatter? If so, skip this comment (it'd mean that formatter has an error).

Reply:
Yes, it's auto-formatter. Check [this commit](https://github.com/enso-org/enso/pull/3588/commits/99b438e16299a914af5fe9b71177f1d0615c6f24)